airbyte-integrations/connectors/destination-mongodb-strict-encrypt/src/main/java/io.airbyte.integrations.destination.mongodb/MongodbDestinationStrictEncrypt.java

```diff
@@ -21,6 +25,19 @@ public MongodbDestinationStrictEncrypt() {
     super(MongodbDestination.sshWrappedDestination());
   }
 
+  @Override
+  public AirbyteConnectionStatus check(final JsonNode config) throws Exception {
+    final JsonNode instanceConfig = config.get(MongoDbDestinationUtils.INSTANCE_TYPE);
+    final MongoInstanceType instance = MongoInstanceType.fromValue(instanceConfig.get(MongoDbDestinationUtils.INSTANCE).asText());
+    // If the MongoDb destination connector is not set up to use a TLS connection, then check should fail
+    if (instance.equals(MongoInstanceType.STANDALONE) && !MongoDbDestinationUtils.tlsEnabledForStandaloneInstance(config, instanceConfig)) {
+      return new AirbyteConnectionStatus()
```

Comment:
If this isn't urgent - can you hold off until I merge the PR that adds the config exception handling? That way - you can just throw a ConfigErrorException instead

Reply:
Sure, I'll probably rework the `MongodbDestinationUtils` and `MongodbSourceUtils` into `MongoUtils.java` in the meantime. Your changes for that PR config exception should be ready to merge unless there's something else you needed to follow up on

Reply:
Yeah, there's still a degree of integration test mangling that I need to do for that PR. To unblock you and others I have https://github.com/airbytehq/airbyte/pull/18989 (which you already commented on)

I'll going to break that PR into smaller parts, but overall you should be unblocked on that front soon

Reply:
That PR is merged, so you should be able to throw a ConfigErrorException here now